src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -2865,31 +2865,49 @@ private void EnsureForwardTypeToAssemblyMap()
         {
             if (_lazyForwardedTypesToAssemblyMap == null)
             {
-                var typesToAssemblyMap = new Dictionary<string, AssemblyReferenceHandle>();
+                var typesToAssemblyMap = new Dictionary<string, HashSet<AssemblyReferenceHandle>>();
 
                 try
                 {
                     var forwarders = MetadataReader.ExportedTypes;
                     foreach (var handle in forwarders)
                     {
                         ExportedType exportedType = MetadataReader.GetExportedType(handle);
+
                         if (!exportedType.IsForwarder)
                         {
                             continue;
                         }
 
+                        AssemblyReferenceHandle newReference = (AssemblyReferenceHandle)exportedType.Implementation;
+
+                        if (newReference.IsNil)
+                        {
+                            continue;
```

Comment:
So we have two options in case of invalid metadata, either discard it, or keep the null references. I chose to discard invalid values for now. If we have a repro for invalid IL that we need to handle in a different way, I'll open another bug for that. But for now, I chose to discard it.